libs/net/services/Actions/Managers/DataSourceIngestManager`.cs

```diff
@@ -58,15 +58,15 @@ protected override async Task<bool> PreRunAsync()
         if (dataSource != null)
             this.DataSource = dataSource;
 
-        return VerifyDataSource() && this.DataSource.DataSourceSchedules.Where(s => s.Schedule != null).Any(s => VerifySchedule(GetSourceDateTime(DateTime.UtcNow), s.Schedule!));
+        return VerifyDataSource() && this.DataSource.DataSourceSchedules.Where(s => s.Schedule != null).Any(s => VerifySchedule(s.Schedule!));
     }
 
     /// <summary>
     /// Get the date and time for the source timezone.
     /// </summary>
     /// <param name="date"></param>
     /// <returns></returns>
-    protected virtual DateTime GetSourceDateTime(DateTime date)
+    public virtual DateTime GetSourceDateTime(DateTime date)
```

Comment:
Actions needed access to these methods.  Probably should convert to an extension method.